hr_switzerland/models/hr_attendance.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2018 Compassion CH (http://www.compassion.ch)
+#    @author: Stephane Eicher <seicher@compassion.ch>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+
+from odoo import models, api
+
+
+class HrAttendance(models.Model):
+    _inherit = 'hr.attendance'
+
+    ##########################################################################
+    #                               ORM METHODS                              #
+    ##########################################################################
+    @api.model
+    def create(self, vals):
+        """ Log the agent related to the user during the checking """
+        new_record = super(HrAttendance, self).create(vals)
+        user = new_record.employee_id.user_id
+        if 'check_in' in vals and user.resource:
+            self.env['hr.asterisk_autolog'].agent_connexion(ast_user=user,
+                                                            connect=True)
+
+        return new_record
+
+    @api.multi
+    def write(self, vals):
+        """ Logout the agent related to the user during the checking """
+        user = self.employee_id.user_id
+        if (len(self) == 1) and ('check_out' in vals) and user.resource:
+            # todo: not enough check...
+            self.env['hr.asterisk_autolog'].agent_connexion(ast_user=user,
```

Comment:
I think this will potentially disconnect users if they modifiy previous attendance. Why not using the field `attendance_state` in `hr.employee` to check if the user is logged in or not?